contracts/strategies/venus/VenusVAIStrategyMainnet.sol

```diff
@@ -8,44 +8,57 @@ import "@pancakeswap/pancake-swap-lib/contracts/token/BEP20/SafeBEP20.sol";
 import "@openzeppelin/contracts-upgradeable/math/MathUpgradeable.sol";
 import "@pancakeswap/pancake-swap-lib/contracts/math/SafeMath.sol";
 
-import "./interface/IVAIVault.sol";
-import "../../base/StrategyBase.sol";
+import "../../base/venus-base/interface/IVAIVault.sol";
+import "../../base/upgradability/BaseUpgradeableStrategy.sol";
 import "../../base/interface/pancakeswap/IPancakeRouter02.sol";
 
-contract VenusVAIStrategy is StrategyBase {
+contract VenusVAIStrategyMainnet is BaseUpgradeableStrategy {
 
   using SafeMath for uint256;
   using SafeBEP20 for IBEP20;
 
-  bool pausedInvesting = false; // When this flag is true, the strategy will not be able to invest. But users should be able to withdraw.
-  IVAIVault public rewardPool = IVAIVault(address(0x0667Eed0a0aAb930af74a3dfeDD263A73994f216));
-  address public wbnb = address(0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c);
-  address public busd = address(0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56);
-  address public vai = address(0x4BD17003473389A42DAF6a0a729f6Fdb328BbBd7);
-  address public venus = address(0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63);
-
+  IVAIVault public rewardPoolContract;
+  address public pancakeRouterV2;
   address[] public liquidationPath;
 
-  uint256 public sellFloor = 1e16;
+  // These tokens cannot be claimed by the controller
+  mapping(address => bool) public unsalvagableTokens;
```

Comment:
For upgradable strategies, we typically define this as a function that returns `true if the input is `rewardToken() or `underlying()`